Tests/test_SeqIO.py

```diff
@@ -369,357 +125,3284 @@ def alignment_summary(alignment, index=" "):
+                bad = protein_alphas + rna_alphas
+            elif isinstance(base_alpha, Alphabet.NucleotideAlphabet):
+                good = nucleotide_alphas
+                bad = protein_alphas
+            else:
+                self.assertIn(t_format, no_alpha_formats,
+                    "Got %s from %s file" % (repr(base_alpha), t_format))
+                good = protein_alphas + dna_alphas + rna_alphas + nucleotide_alphas
+                bad = []
+            for given_alpha in good:
+                # These should all work...
+                given_base = Alphabet._get_base_alphabet(given_alpha)
+                for record in SeqIO.parse(t_filename, t_format, given_alpha):
+                    base_alpha = Alphabet._get_base_alphabet(record.seq.alphabet)
+                    self.assertIsInstance(base_alpha, given_base.__class__)
+                    self.assertEqual(base_alpha, given_base)
+                if t_count == 1:
+                    h = open(t_filename, mode)
+                    record = SeqIO.read(h, t_format, given_alpha)
+                    h.close()
+                    self.assertIsInstance(base_alpha, given_base.__class__)
+                    self.assertEqual(base_alpha, given_base)
+            for given_alpha in bad:
+                # These should all fail...
+                with open(t_filename, mode) as h:
+                    records6 = SeqIO.parse(h, t_format, given_alpha)
+                    self.assertRaises(ValueError, next, records6,
+                        "Forcing wrong alphabet, %s, should fail (%s)"
+                                     % (repr(given_alpha), t_filename))
+            del good, bad, given_alpha, base_alpha
+
+            if t_alignment:
+                alignment = MultipleSeqAlignment(SeqIO.parse(
+                        handle=t_filename, format=t_format))
+                self.assertEqual(len(alignment), t_count)
+                alignment_len = alignment.get_alignment_length()
+
+                # Check the record order agrees, and double check the
+                # sequence lengths all agree too.
+                for i in range(t_count):
+                    self.assertEqual(records[i], alignment[i])
+                    self.assertEqual(len(records[i].seq), alignment_len)
+
+                self.assertEqual(str(alignment_summary(alignment)),
+                                 expected_alignment)
+
+        # Some alignment file formats have magic characters which mean
+        # use the letter in this position in the first sequence.
+        # They should all have been converted by the parser, but if
+        # not reversing the record order might expose an error.  Maybe.
+        records.reverse()
+        self.check_simple_write_read(records, t_format, t_count,
+                                     expected_messages)
+
+    def test_sff1(self):
+        records = [SeqRecord(Seq("tcagGGTCTACATGTTGGTTAACCCGTACTGATTTGAATT...GGGCTTa"), "E3MFGYR02JWQ7T", "E3MFGYR02JWQ7T"),
```

Comment:
I find it rather misleading that you make a list of expected ``SeqRecord`` objects but with truncated sequences. I would find it clearer to have a list of expected identifiers (strings in full) and truncated sequences - to match the list of expected sequence lengths.

Reply:
Update looks clearer to me now, thanks!

Reply:
OK to proceed? The remaining AppVeyor failure is unrelated to `test_SeqIO.py`.